DigitalSolutionsStudio.Api/D-Studio. Test task C# code sample.cs

```diff
@@ -0,0 +1,121 @@
+    }
+    else if (cboHistoryType.Text.Equals("Tick Timeframe"))
+    {
+        // request in the format:
+        // HTT,SYMBOL,BEGINDATE BEGINTIME,ENDDATE ENDTIME,MAXDATAPOINTS,BEGINFILTERTIME,ENDFILTERTIME,DIRECTION,REQUESTID,DATAPOINTSPERSEND<CR><LF>
+        sRequest = String.Format("HTT,{0},{1},{2},{3},{4},{5},{6},{7},{8}\r\n", txtSymbol.Text, txtBeginDateTime.Text, txtEndDateTime.Text, txtDatapoints.Text, txtBeginFilterTime.Text, txtEndFilterTime.Text, txtDirection.Text, txtRequestID.Text, txtDatapointsPerSend.Text);
+    }
+    else if (cboHistoryType.Text.Equals("Interval Datapoints"))
+    {
+        // validate interval type
+        string sIntervalType = "s";
+        if (rbVolume.Checked)
+        {
+            sIntervalType = "v";
+        }
+        else if (rbTick.Checked)
+        {
+            sIntervalType = "t";
+        }
+
+        // request in the format:
+        // HIX,SYMBOL,INTERVAL,NUMDATAPOINTS,DIRECTION,REQUESTID,DATAPOINTSPERSEND,INTERVALTYPE<CR><LF>
+        sRequest = String.Format("HIX,{0},{1},{2},{3},{4},{5},{6},{7}\r\n", txtSymbol.Text, txtInterval.Text, txtDatapoints.Text, txtDirection.Text, txtRequestID.Text, txtDatapointsPerSend.Text, sIntervalType, chkBoxTimeStamp.Checked ? 1 : 0);
+    }
+    else if (cboHistoryType.Text.Equals("Interval Days"))
+    {
+        // validate interval type
+        string sIntervalType = "s";
+        if (rbVolume.Checked)
+        {
+            sIntervalType = "v";
+        }
+        else if (rbTick.Checked)
+        {
+            sIntervalType = "t";
+        }
+
+        // request in the format:
+        // HID,SYMBOL,INTERVAL,NUMDAYS,MAXDATAPOINTS,BEGINFILTERTIME,ENDFILTERTIME,DIRECTION,REQUESTID,DATAPOINTSPERSEND,INTERVALTYPE,TIMESTAMPLABEL<CR><LF>
+        sRequest = String.Format("HID,{0},{1},{2},{3},{4},{5},{6},{7},{8},{9},{10}\r\n", txtSymbol.Text, txtInterval.Text, txtDays.Text, txtDatapoints.Text, txtBeginFilterTime.Text, txtEndFilterTime.Text, txtDirection.Text, txtRequestID.Text, txtDatapointsPerSend.Text, sIntervalType, chkBoxTimeStamp.Checked ? 1 : 0);
+    }
+    else if (cboHistoryType.Text.Equals("Interval Timeframe"))
+    {
+        // validate interval type
+        string sIntervalType = "s";
+        if (rbVolume.Checked)
+        {
+            sIntervalType = "v";
+        }
+        else if (rbTick.Checked)
+        {
+            sIntervalType = "t";
+        }
+
+        // request in the format:
+        // HIT,SYMBOL,INTERVAL,BEGINDATE BEGINTIME,ENDDATE ENDTIME,MAXDATAPOINTS,BEGINFILTERTIME,ENDFILTERTIME,DIRECTION,REQUESTID,DATAPOINTSPERSEND,INTERVALTYPE,TIMESTAMPLABEL<CR><LF>
+        sRequest = String.Format("HIT,{0},{1},{2},{3},{4},{5},{6},{7},{8},{9},{10},{11}\r\n", txtSymbol.Text, txtInterval.Text, txtBeginDateTime.Text, txtEndDateTime.Text, txtDatapoints.Text, txtBeginFilterTime.Text, txtEndFilterTime.Text, txtDirection.Text, txtRequestID.Text, txtDatapointsPerSend.Text, sIntervalType, chkBoxTimeStamp.Checked ? 1 : 0);
```

Comment:
"chkBoxTimeStamp.Checked ? 1 : 0" should be refactored by using a separate constants file, which will contain const's with appropriate and telling names.